test/integration/resourcemanager/podschedulername/podschedulername_suite_test.go

```diff
@@ -77,27 +112,36 @@ var _ = BeforeSuite(func() {
 	Expect(err).NotTo(HaveOccurred())
 
 	By("registering webhook")
-	conf := podschedulername.WebhookConfig{SchedulerName: "bin-packing-scheduler"}
-	Expect(podschedulername.AddToManagerWithOptions(mgr, conf)).To(Succeed())
+	Expect(podschedulername.AddToManagerWithOptions(mgr, podschedulername.WebhookConfig{
+		SchedulerName: "bin-packing-scheduler",
+	})).To(Succeed())
 
 	By("starting manager")
-	var mgrContext context.Context
-	mgrContext, mgrCancel = context.WithCancel(ctx)
+	mgrContext, mgrCancel := context.WithCancel(ctx)
+
 	go func() {
 		defer GinkgoRecover()
 		Expect(mgr.Start(mgrContext)).To(Succeed())
 	}()
-})
 
-var _ = AfterSuite(func() {
-	By("stopping manager")
-	mgrCancel()
-
-	By("stopping test environment")
-	Expect(testEnv.Stop()).To(Succeed())
+	DeferCleanup(func() {
+		By("stopping manager")
+		mgrCancel()
+	})
 })
 
-func getMutatingWebhookConfigurations() []*admissionregistrationv1.MutatingWebhookConfiguration {
+func getMutatingWebhookConfigurations(namespaceName string) []*admissionregistrationv1.MutatingWebhookConfiguration {
+	webhook := resourcemanager.GetPodSchedulerNameMutatingWebhook(nil, func(_ *corev1.Secret, path string) admissionregistrationv1.WebhookClientConfig {
```

Comment:
We could adapt the `GetPodSchedulerNameMutatingWebhook` func to accept the namespaceSelector as first arg (similar to the `GetTokenInvalidatorMutatingWebhook` and `getProjectedTokenMountMutatingWebhook` funcs).